### PR TITLE
[Feat]: watchFocus option

### PR DESCRIPTION
**Pull request by S-Shingler**
Hi David,

Firstly, thank you for an awesome package!

I went more in depth regarding my issue in this discussion:

- https://github.com/davidjerleke/embla-carousel/discussions/936

Essentially I need a way to override the default slide focus behaviour for some accessibility work I've been doing.

A `watchFocus` option feels natural to me but curious to hear your thoughts.

Cheers,
Sam

---

### Closes:

- #936
- #764
- #642
- #822
- #854

### CODE REVIEW

File: packages/embla-carousel/src/components/SlideFocus.ts [hunk 1]
```diff
@@ -16,11 +20,14 @@ export function SlideFocus(
...
   scrollBody: ScrollBodyType,
   eventStore: EventStoreType,
-  eventHandler: EventHandlerType
+  eventHandler: EventHandlerType,
+  watchFocus: FocusHandlerOptionType
 ): SlideFocusType {
   let lastTabPressTime = 0
 
   function init(): void {
+    if (!watchFocus) return
```
Comment (Mitch-At-Work) [inline packages/embla-carousel/src/components/SlideFocus.ts:29]: +1 for this, we are encountering some accessibility issues with our setup from focus causing movement - especially in groups that are un-even.

**Comment by davidjerleke** [comment]
Thanks for your contribution @S-Shingler 👍🏻,

I’m on vacation so I will review and test this when I’m back. Thank you for your patience.

**Comment by S-Shingler** [comment]
No worries, enjoy your vacation!

**Comment by davidjerleke** [comment]
Hi @Mitch-At-Work,

Thanks for your input. I think it's better to create a separate [discussion](https://github.com/davidjerleke/embla-carousel/discussions/new?category=help) if you're unsure about how the current focus behavior is operating. By all means, what you're mentioning could be a bug but it could also be by design and you might be misunderstanding the feature. 

I think it makes sense to not aim for just opting out of this feature if it's causing you trouble, but rather creating a discussion with a CodeSandbox along with clear steps to reproduce the behavior you think is problematic. This way, if it turns out to be a bug, we can fix the bug and improve the library. If not, you might understand why it works the way it does.

With that said, this feature is on the roadmap and I will review and test this PR as soon as possible so it's upcoming:

- #841

**Comment by davidjerleke** [comment]
### Note to self:

Test focus behavior for:

- AutoScroll with links #764.
- Fade.

**Comment by davidjerleke** [comment]
Hi @S-Shingler,

I've made some modifications in order to align the `watchFocus` option structure with the other watchers like `watchResize`, `watchDrag` and `watchSlides`. I think this feature is ready for a release but I would appreciate if you would try it out before we do so I haven't missed anything obvious.

In you local environment, you can try any of the three playgrounds and test the new feature out:

- `yarn start:vanilla` (starts the vanilla playground)
- `yarn start:react` (starts the react playground)
- `yarn start:solid` (starts the solid playground)

All environments live in this folder: `<your_root_folder>/playgrounds/`. Let me know how it goes!

Thanks in advance.

**Comment by S-Shingler** [comment]
Hi @davidjerleke,

Everything is looking good.

Cheers,
Sam

**Comment by davidjerleke** [comment]
@S-Shingler this feature has been included in the [v8.2.0 release](https://github.com/davidjerleke/embla-carousel/releases/tag/v8.2.0).

**Comment by S-Shingler** [comment]
> @S-Shingler, @Mitch-At-Work, this feature has been included in the [v8.2.0 release](https://github.com/davidjerleke/embla-carousel/releases/tag/v8.2.0). Thanks a lot for your contribution 💫! 

That's awesome, thank you! @davidjerleke

**Pull request merged**